images/clustermesh-apiserver/Dockerfile

```diff
@@ -1,15 +1,16 @@
 # Copyright Authors of Cilium
 # SPDX-License-Identifier: Apache-2.0
 
-ARG BASE_IMAGE=scratch
+# distroless images are signed by cosign and can be verified using:
+# cosign verify $IMAGE_NAME --certificate-oidc-issuer https://accounts.google.com  --certificate-identity keyless@distroless.iam.gserviceaccount.com
+ARG BASE_IMAGE=gcr.io/distroless/static-debian11:latest@sha256:6d31326376a7834b106f281b04f67b5d015c31732f594930f2ea81365f99d60c
```

Comment:
Why is this a different sha than the upstream commit?

Reply:
> Replaced the nonroot base image with the root one, to avoid requiring the Helm changes to configure the fsGroup, which could cause issues if users only updated the image version, without a full helm upgrade.

I'm personally concerned that backporting the change to the nonroot image (which then makes the clustermesh-apiserver containers to run as a non root user) would be potentially problematic on a patch version, as modifying the behavior and requiring helm changes as well (the apiserver container is also starting by default an http server on a privileged port for health checks, although never used). The usage of the one with the root user (named latest) instead allows to benefit from the /tmp directory, without changing the UID which is used.

Does that make sense to you?

Reply:
Yeah makes sense, thanks.